docs/2.0/docs/pipelines/guides/stacks.md

```diff
@@ -0,0 +1,42 @@
+# Running Pipelines with Terragrunt Stacks
+
+## Introduction
+
+Terragrunt Stacks provide a powerful way to organize and manage your infrastructure code. Gruntwork Pipelines automatically integrates with Terragrunt Stacks.
+
+Learn more about Terragrunt Stacks in the <span class="external-link"><a href="https://terragrunt.gruntwork.io/docs/features/stacks/">Terragrunt Stacks Documentation</a></span>.
+
+## Prerequisites
+
+Pipelines automatically supports Terragrunt Stacks as long as you're using Terragrunt `v0.71.3` or higher. You can specify the Terragrunt binary version in the `mise.toml` file in your repository, see the [configuration reference](/2.0/reference/pipelines/configurations#example-mise-configuration).
+
+See [Terragrunt Version Compatibility](/2.0/reference/pipelines/terragrunt-version-compatibility) for more details on specific Terragrunt versions and their compatibility with Pipelines.
+
+## Terragrunt Stacks in Pipelines
+
+### Detecting Stack Changes
+
+When you create, update, or remove a `terragrunt.stack.hcl` file, Pipelines will automatically:
+
+1. Generate the stack during orchestration 
+2. Detect any changes to units and values defined in the stack
+3. Plan and apply the changes as part of your existing pipelines process
+
+Note that the stack will only be generated if there are changes to the stack file. If you make changes to the remote units defined in the stack, these changes will not be reflected in pipeline runs until the stack file is updated.
+
+### Repository Structure
+
+We recommend using your `infrastructure-catalog` repository to define your reusable Terragrunt Units.
+
+Within your `infrastructure-live-root` repositories, add `terragrunt.stack.hcl` files to define your Terragrunt Stacks. You should not commit the generated `.terragrunt-stack` directories to source control, pipelines will generate these directories automatically during execution.
+
```

Comment:
Please consider adding a level 4 header to this section so the preflight error can link directly to it.

Reply:
Done!